### PR TITLE
Te 652 rikerout

### PR DESCRIPTION
**Pull request by bkontur**

### CODE REVIEW

File: light_node/src/main.rs [hunk 1]
```diff
@@ -364,6 +364,7 @@ fn block_on_actors(
             shell_compatibility_version,
             env.p2p,
             env.identity.expected_pow,
+            Arc::new(init_storage_data.chain_id),
```
Comment (binier) [inline light_node/src/main.rs:367]: Why `Arc` ? Do we need chain_id in `PeerManager`?

**Review by binier — CHANGES_REQUESTED** [review]
Please check comments and rebase your branch to latest **p2p-simulator** branch.

File: light_node/src/main.rs (hunk 1 above)
Comment (bkontur) [reply light_node/src/main.rs]: @tizoc 
arc removed, chain_id is needed in tezedge_state

**Comment by bkontur** [comment]
@binier @jurajselep 
Zura, could you please check this PR, when ok, I would like to merge it to p2p-simulator branch, there is:
- a removed peer_manager actor + changed light-node initialization of TezedgeStateManager (state + proposer)
- removed usage of network_channel for several shell actors (chain_manager, peer_branch_bootstrapper) replaced by proposer

this is something, which you have to do anyway and it makes other things easier for both

File: shell/src/peer_manager.rs [hunk 2]
```diff
@@ -91,17 +95,26 @@ impl From<std::io::Error> for NotifyProposerError {
     }
 }
 
+impl From<std::sync::mpsc::SendError<ProposerMsg>> for NotifyProposerError {
+    fn from(err: std::sync::mpsc::SendError<ProposerMsg>) -> Self {
+        Self::SendError(err)
+    }
+}
+
+pub type ProposerHandleRef = ProposerHandle;
```
Comment (binier) [inline shell/src/peer_manager.rs:104]: Why the alias?

**Review by binier — CHANGES_REQUESTED** [review]
PeerManager actor still is here. Can we get rid of it? Doesn't look like it's used anywhere.

File: shell/src/peer_manager.rs (hunk 2 above)
Comment (bkontur) [reply shell/src/peer_manager.rs:104]: @binier 
peer_manager is not used anywhere, I just did not want to remove it to aviod rebase conflicts,
but when you approve this, I will clean it and merge to your branch

Comment (bkontur) [reply shell/src/peer_manager.rs:104]: @binier 
to that alias, I had there before:
```
pub type ProposerHandleRef = Arc<ProposerHandle>;
```
not needed, removing

Comment (bkontur) [reply shell/src/peer_manager.rs]: @binier 
removed all peer_manger riker actor related stuff + removed ProposerHandleRef

**Review by binier — APPROVED** [review]
Looks good! Please rebase to latest **p2p-simulator**, let's make sure tests pass and we can merge it.

**Comment by bkontur** [comment]
@binier 
ok, thx, merged,

tests passed, except these two:
- benchmark for stream decoding was removed (we should rework it or remove from drone)
-- http://ci.tezedge.com/tezedge/tezedge/2040/34/7
- bootstrap time test 
-- http://ci.tezedge.com/tezedge/tezedge/2040/34/7
-- we have now slower bootstrap with state machine (I guess, once we are done with refactor, it will be fixed or even better)

but both are also problem for `origin/p2p-simulator` branch:
http://ci.tezedge.com/tezedge/tezedge/2024/34/7
http://ci.tezedge.com/tezedge/tezedge/2024/29/5
so they are not related to this PR

**Pull request merged**